src/Microsoft.AspNet.Server.Kestrel/Http/DateHeaderValueManager.cs

```diff
@@ -0,0 +1,145 @@
+            TimeSpan timerInterval)
+        {
+            _systemClock = systemClock;
+            _timeWithoutRequestsUntilIdle = timeWithoutRequestsUntilIdle;
+            _timerInterval = timerInterval;
+        }
+
+        /// <summary>
+        /// Returns a value representing the current server date/time for use in the HTTP "Date" response header
+        /// in accordance with http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.18
+        /// </summary>
+        /// <returns>The value.</returns>
+        public virtual string GetDateHeaderValue()
+        {
+            PumpTimer();
+
+            // See https://msdn.microsoft.com/en-us/library/az4se3k1(v=vs.110).aspx#RFC1123 for info on the format
+            // string used here.
+            // The null-coalesce here is to protect against returning null after Dispose() is called, at which
+            // point _dateValue will be null forever after.
+            return _dateValue ?? _systemClock.UtcNow.ToString("r");
+        }
+
+        /// <summary>
+        /// Releases all resources used by the current instance of <see cref="DateHeaderValueManager"/>.
+        /// </summary>
+        public void Dispose()
+        {
+            lock (_timerLocker)
+            {
+                if (_dateValueTimer != null)
+                {
+                    DisposeTimer();
+                }
+
+                _isDisposed = true;
+            }
+        }
+
+        private void PumpTimer()
+        {
+            _hadRequestsSinceLastTimerTick = true;
+
+            // If we're already disposed we don't care about starting the timer again. This avoids us having to worry
+            // about requests in flight during dispose (not that that should actually happen) as those will just get
+            // SystemClock.UtcNow (aka "the slow way").
+            if (!_isDisposed && _dateValueTimer == null)
+            {
+                lock (_timerLocker)
+                {
+                    if (!_isDisposed && _dateValueTimer == null)
+                    {
+                        // Immediately assign the date value and start the timer again. We assign the value immediately
+                        // here as the timer won't fire until the timer interval has passed and we want a value assigned
+                        // inline now to serve requests that occur in the meantime.
+                        _dateValue = _systemClock.UtcNow.ToString("r");
+                        _dateValueTimer = new Timer(UpdateDateValue, state: null, dueTime: _timerInterval, period: _timerInterval);
+                    }
+                }
+            }
+        }
+
+        // Called by the Timer (background) thread
+        private void UpdateDateValue(object state)
+        {
+            var now = _systemClock.UtcNow;
+
+            // See http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.18 for required format of Date header
+            _dateValue = now.ToString("r");
```

Comment:
If you're not going to do that, at least move `http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.18` to a resource string :trollface: